### PR TITLE
fix: parse method mutating its input

### PR DESCRIPTION
**Pull request by matejvasek**
fixes: #230

### CODE REVIEW

File: src/lib/bindings/http/validation/structured.js [hunk 1]
```diff
@@ -32,17 +32,20 @@ function parse(payload, headers, receiver) {
     source: undefined
   };
 
+
+  const unprocessed = new Set(Object.keys(incoming));
```
Comment (lholmquist) [inline src/lib/bindings/http/validation/structured.js:36]: is `Set` necessary here.   Can't we assume that the keys in the object are unique already?

Comment (matejvasek) [reply src/lib/bindings/http/validation/structured.js:36]: I think they should be unique. The thing is: If I understand it correctly `Object.keys(incoming)` returns an array and deletion from an array is more complicated than just `delete()` which is offered by a `Set` object.

Comment (lance) [inline src/lib/bindings/http/validation/structured.js:36]: I think the most efficient would be using the spread operator like this.
```suggestion
  const unprocessed = { ...incoming };
```

**Comment by matejvasek** [comment]
@lance I force pushed it. There is now only one line changed: `const incoming = { ...parser.parse(payload) };`.

**Pull request merged**